tree/ntuple/v7/src/RPageStorageFile.cxx

```diff
@@ -87,6 +94,16 @@ ROOT::Experimental::Detail::RPageSinkFile::~RPageSinkFile()
 {
 }
 
+int ROOT::Experimental::Detail::RPageSinkFile::UpdateCompression() {
+   if (fOptions.GetCompression() != RCompressionSetting::ELevel::kInherit) {
+      return fOptions.GetCompression();
+   }
+   auto compression = fWriter->GetTFileCompression();
+   for (auto &columnRange : RPageSink::fOpenColumnRanges) {
```

Comment:
Let's add a check if the compression actually changed since last time we checked.

Reply:
Updated in 07384b2